Realm/ObjectServerTests/RLMObjectServerTests.mm

```diff
@@ -1451,6 +1451,22 @@ - (void)testDownloadWhileOpeningRealm {
     }
 }
 
+- (void)testDownloadCancelsOnAuthError {
+    RLMSyncUser *user = [self logInUserForCredentials:[RLMObjectServerTests basicCredentialsWithName:NSStringFromSelector(_cmd)
+                                                                                            register:self.isParent]
+                                               server:[RLMObjectServerTests authServerURL]];
+    auto c = [RLMRealmConfiguration defaultConfiguration];
+    c.syncConfiguration = [[RLMSyncConfiguration alloc] initWithUser:user realmURL:[NSURL URLWithString:@"realm://localhost:9080/invalid"]];
```

Comment:
I'm likely missing something, but why is that path invalid?

Reply:
The user shouldn't have permissions to open a realm at that path.

Reply:
Ah... right, I was trying to relate the test to the other changes, but I guess they are not directly related.

Reply:
A permission denied error hits the new code added in RLMSyncSessionRefreshHandle.mm. The new error code used to be relevant to this, but it turns out that with ROS 3.0 leaving out the realm path gives you the default realm so it's not actually an error.